src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -668,15 +693,39 @@ private static DkmEvaluationResult CreateEvaluationResultFromException(Exception
                 DataItem: null);
         }
 
-        internal static void ContinueWithExceptionHandling(CompletionRoutine onCompleted, CompletionRoutine<Exception> onException)
+        private sealed class WorkList
```

Comment:
It would be good to assert that _completionRoutine is null although it's not clear when we would assert since the place to call Dispose is also a reasonable point to call Execute.